routes/resources.js

```diff
@@ -10,27 +10,44 @@ var Schemas = require('../generators/schemas');
 
 module.exports = function (app, model, Implementation, integrator, opts) {
   var modelName = Implementation.getModelName(model);
-  var schema = Schemas.schemas[Implementation.getModelName(model)];
+  var schema = Schemas.schemas[modelName];
+
+  function addSmartFieldToRecord(record, smartFieldObject, modelName) {
```

Comment:
About the function name, this is not exactly what the function does. As you iterate over all fields, it does not necessarily add a Smart Field to the record. It does if the field is a Smart Field.